### PR TITLE
Copy git submodules as-is to avoid auth and path errors

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Connect https://github.com/ansible/awx/issues/4703

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```
7.0.0
```

##### ADDITIONAL INFORMATION
Reading the docs more carefully, this cause of this issue is clear. I know what I want it to copy, and it doesn't offer that option.

Confirmed via manual testing that this fixes the problem for the test case of https://github.com/alancoding/has-relative-submodule. As of this moment, I don't actually know whether this would or wouldn't work for nested submodules. I'm quite confident that this would work for submodules that require auth to their remotes (everything here is local).

This diff has to move code because you can't list submodules for a branch which is not checked out. That's not a problem, because we _do_ have the branch/commit checked out while the project update lock is held. So the copy logic has to be moved there.

### CODE REVIEW

**Comment by AlanCoding** [comment]
recheck

**Comment by AlanCoding** [comment]
recheck

File: awx/main/tasks.py [hunk 1]
```diff
@@ -2065,15 +2047,53 @@ def pre_run_hook(self, instance, private_data_dir):
                 git_repo = git.Repo(project_path)
                 self.original_branch = git_repo.active_branch
 
+    @staticmethod
+    def make_local_copy(project_path, destination_folder, scm_type, scm_revision):
+        if scm_type == 'git':
+            git_repo = git.Repo(project_path)
+            if not os.path.exists(destination_folder):
+                os.mkdir(destination_folder)
```
Comment (ryanpetrello) [inline awx/main/tasks.py]: Should we set a reasonable chmod here? e.g., 

```python
os.mkdir(..., stat.S_IREAD | stat.S_IWRITE | stat.S_IEXEC)
```

File: awx/main/tasks.py [hunk 2]
```diff
@@ -2065,15 +2047,53 @@ def pre_run_hook(self, instance, private_data_dir):
...
+            )
+            # submodules copied in loop because shallow copies from local HEADs are ideal
+            # and no git clone submodule options are compatible with minimum requirements
+            for submodule in git_repo.submodules:
+                subrepo_path = os.path.abspath(os.path.join(project_path, submodule.path))
+                subrepo_destination_folder = os.path.abspath(os.path.join(destination_folder, submodule.path))
+                subrepo_uri = Path(subrepo_path).as_uri()
+                # may need to include branch, first have to test this
+                git.Repo.clone_from(subrepo_uri, subrepo_destination_folder, depth=1, single_branch=True)
+            # TODO?: copy submodules of submodules
```
Comment (ryanpetrello) [inline awx/main/tasks.py]: Eh.

**Review by ryanpetrello — APPROVED** [review]

**Pull request merged**